File: Date/CS/Date.cs
using System;

namespace NumbatLogic
{
    class Date
    {
        public static uint GetUnixDate()
        {
            long value = DateTimeOffset.UtcNow.ToUnixTimeSeconds();
            if (value < 0)
                return 0;
            if (value > uint.MaxValue)
                return uint.MaxValue;
            return (uint)value;
        }
    }
}

File: Date/CPP/Date.cpp
#include "Date.hpp"

#include <chrono>

namespace NumbatLogic
{
	uint32_t Date::GetUnixDate()
	{
		auto now = std::chrono::system_clock::now();
		auto epoch = std::chrono::system_clock::from_time_t(0);
		auto diff = std::chrono::duration_cast<std::chrono::seconds>(now - epoch);
		auto value = diff.count();
		if (value < 0)
			return 0;
		if (value > 0xFFFFFFFFull)
			return 0xFFFFFFFFu;
		return static_cast<uint32_t>(value);
	}
}

